wsaa.py

```diff
@@ -107,12 +107,53 @@ def create_tra(service=SERVICE, ttl=2400):
     return tra.as_xml()
 
 
-def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
-    "Firmar PKCS#7 el TRA y devolver CMS (recortando los headers SMIME)"
+def sign_tra_new(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    "Sign Digital transactions with cryptography versions >= 39 in python 3"
+    # Leer privatekey y cert
+    if not privatekey.startswith(b"-----BEGIN RSA PRIVATE KEY-----"):
+        privatekey = open(privatekey).read()
+        if isinstance(privatekey, str):
+            privatekey = privatekey.encode("utf-8")
+
+    if not passphrase:
+        password = None
+    else:
+        password = passphrase
+    private_key = serialization.load_pem_private_key(
+        privatekey, password, default_backend()
+    )
+
+    if not cert.startswith(b"-----BEGIN CERTIFICATE-----"):
+        cert = open(cert).read()
+        if isinstance(cert, str):
+            cert = cert.encode("utf-8")
+    cert = x509.load_pem_x509_certificate(cert)
+
+    p7 = pkcs7.PKCS7SignatureBuilder().set_data(
+        tra
+    ).add_signer(
+        cert, private_key, hashes.SHA256()
+    ).sign(
+        serialization.Encoding.SMIME, [pkcs7.PKCS7Options.Binary]
+    )
+
+    # Generar p7 en formato mail y recortar headers
+    msg = email.message_from_string(p7.decode("utf8"))
+    for part in msg.walk():
+        filename = part.get_filename()
+        if filename and filename.startswith("smime.p7"):
+            # Es la parte firmada?
+            # Devolver CMS
+            return part.get_payload(decode=False)
+    else:
+        raise RuntimeError("Part not found")
 
-    if isinstance(tra, str):
-        tra = tra.encode("utf8")
 
+
+def sign_tra_old(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    """Legacy method for signing python 2.7 digital transactions on python 2.7 
+    and python 3 versions that don't support cryptography version >39
+    """
     if Binding:
```

Comment:
This condition should be removed (and unindent), it already is at top level `sign_tra()` function